src/js/api/api.js

```diff
@@ -648,6 +648,29 @@ export default function Api(element) {
             return this;
         },
 
+        /**
+         * Sets playlist item specified by `index`.
+         * @param {number} [index] A 0-based index of the desired playlist item.
+         * @param {PlaylistItem} item The new playlist item.
+         * @returns {Api} The Player API instance.
+         */
+
+        setPlaylistItem(index, item) {
+            const state = this.getState();
+
+            if (state === 'idle' || state === 'complete') {
+                let playlist = this.getPlaylist();
+
+                if (playlist) {
+                    playlist = playlist.slice();
+                    playlist[index] = item;
+                    core.updatePlaylist(playlist, core.get('feedData'));
```

Comment:
Why are you copying the playlist and callling `updatePlaylist`?

I can see how it `filterPlaylist` and `_setItem` would be useful if you were to replace an item with something that could not be played or update the current item, however, there some side effects. `stop()` is called and both a "playlist" and "playlistItem" API event will be dispatched. This has not been discussed and is not functionality we want introduced without going through the impact this could have.

Ideally calling `setPlaylistItem` should update an item without changed to the playlist, or interruption to playback, unless the current item is replaces while not idle or complete. The array should remain the same array, unless the item is invalid (in that case we could make the call fail instead of filtering), and the only part of `filterPlaylist` we need for a single item is the `forEach` function being called on items in the list.